### DISCUSSION TITLE
Switching from synth/midi to kit irreversably transposes notes to kit notes

### DISCUSSION THREAD

**Discussion by o010b** [SynthstromAudible/DelugeFirmware#609] [Ideas]
### Please describe the problem:

If (accidentally) switching the instrument from synth/midi to kit, all current note data is transposed to fit the notes of the kit. Switching back to synth/midi retains the transposed notes. Undo does not work. 

### What is the expected behavior?

Switching back to original note pitches. 

Alternatively, press undo button to reverse transformations.

Alternatively, a warning/confirmation before switching a track mode if the track contains note data. 

Maybe all three.

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

~~community~~ official

### What is the firmware name:

4.1.3

### If possible provide the steps to reproduce the issue and upload additional media:

1. Create a glorious synth track. Wear sunglasses at night. 
2. Butter fingers and excitedly flub the kit button.
3. Sigh of the disappointment of your being and switch back to synth mode.
4. Enjoy experimental atonal cacophony that surely will revolutionize the world of music.

**Comment by m-m-adams** [comment]
What happens on the official firmware?

**Comment by o010b** [comment]
My bad, it is the official firmware, not community edition.

**Comment by m-m-adams** [comment]
Making this a discussion - I'm not sure the best way to handle it and how to convert things - the current behaviour of interpreting the notes as drums is probably used by someone and so we'd need a way to smartly interpret changed rows and edited sequences somehow

**Comment by o010b** [comment]
Well, it only happens once when switching a clip from synth/midi to drums, so I don't know to what extent anybody could be using it. If they have done it already, it's already stored as drum pitches. At least to implement undo should be one hundred percent safe.
